### PR TITLE
Adding a parameter to handle the dataset images' extensions

### PR DESCRIPTION
**Pull request by SalahAdDin**
The aim is to solve this issue: https://github.com/fizyr/keras-retinanet/issues/1259

### CODE REVIEW

File: keras_retinanet/bin/debug.py [hunk 1]
```diff
@@ -181,6 +182,7 @@ def csv_list(string):
     parser.add_argument('--random-transform', help='Randomly transform image and annotations.', action='store_true')
     parser.add_argument('--image-min-side', help='Rescale the image so the smallest side is min_side.', type=int, default=800)
     parser.add_argument('--image-max-side', help='Rescale the image if the largest side is larger than max_side.', type=int, default=1333)
+    parser.add_argument('--image-extension',   help='Declares the dataset images\' extension.', default='.jpg')
```
Comment (hgaiser) [inline keras_retinanet/bin/debug.py]: This argument is only necessary for Pascal VOC, so we should add it to that subparser.

Comment (SalahAdDin) [reply keras_retinanet/bin/debug.py]: In all `bin` script files, right?

File: keras_retinanet/preprocessing/pascal_voc.py [hunk 2]
```diff
@@ -77,7 +77,7 @@ def __init__(
         data_dir,
         set_name,
         classes=voc_classes,
-        image_extension='.jpg',
+        image_extension,
```
Comment (hgaiser) [inline keras_retinanet/preprocessing/pascal_voc.py]: It's not allowed to have a positional argument after optional arguments.

Comment (SalahAdDin) [reply keras_retinanet/preprocessing/pascal_voc.py]: Image extension must be before to classes, right?

Comment (hgaiser) [reply keras_retinanet/preprocessing/pascal_voc.py]:
```python
def foo(a, b=1, c):
	print(a, b, c)

foo(1, 2)
```

```shell
  File "test.py", line 1
    def foo(a, b=1, c):
            ^
SyntaxError: non-default argument follows default argument
```

You can give the `image_extension` a default value, making it an optional argument.

Comment (SalahAdDin) [reply keras_retinanet/preprocessing/pascal_voc.py]: Ok,

File: keras_retinanet/bin/debug.py [hunk 3]
```diff
@@ -94,6 +94,7 @@ def create_generator(args):
         generator = PascalVocGenerator(
             args.pascal_path,
             args.pascal_set,
+            args.image_extension,
```
Comment (hgaiser) [inline keras_retinanet/bin/debug.py]: Without the argument name this is assigned to `classes`.

Comment (SalahAdDin) [reply keras_retinanet/bin/debug.py]: I don't get you.

Comment (hgaiser) [reply keras_retinanet/bin/debug.py]: The arguments for the `PascalVocGenerator` are:
```python
    def __init__(
        self,
        data_dir,
        set_name,
        classes=voc_classes,
        image_extension='.jpg',
        skip_truncated=False,
        skip_difficult=False,
        **kwargs
    ):
```

So the third argument would be `classes`, where you pass `args.image_extension`.

File: keras_retinanet/bin/evaluate.py [hunk 4]
```diff
@@ -54,6 +54,7 @@ def create_generator(args):
         validation_generator = PascalVocGenerator(
             args.pascal_path,
             'test',
+            args.image_extension,
```
Comment (hgaiser) [inline keras_retinanet/bin/evaluate.py]: Same issue here.

File: keras_retinanet/bin/train.py [hunk 5]
```diff
@@ -266,15 +266,17 @@ def create_generators(args, preprocess_image):
     elif args.dataset_type == 'pascal':
         train_generator = PascalVocGenerator(
             args.pascal_path,
-            'trainval',
+            'train',
+            args.image_extension,
```
Comment (hgaiser) [inline keras_retinanet/bin/train.py]: And here.

File: keras_retinanet/bin/train.py [hunk 6]
```diff
@@ -266,15 +266,17 @@ def create_generators(args, preprocess_image):
...
             transform_generator=transform_generator,
             visual_effect_generator=visual_effect_generator,
             **common_args
         )
 
         validation_generator = PascalVocGenerator(
             args.pascal_path,
-            'test',
+            'val',
+            args.image_extension,
```
Comment (hgaiser) [inline keras_retinanet/bin/train.py]: And here.

**Comment by hgaiser** [comment]
I see this also includes the move from `trainval` to `train` and from `test` to `val`. I checked Pascal VOC 2007 and it seems fine to me.

Thanks for your help!

**Comment by SalahAdDin** [comment]
Your welcome, Now, you can see the F1 score, it works now.

**Pull request merged**